.travis.yml

```diff
@@ -40,27 +40,41 @@ matrix:
   include:
   - compiler: gcc
     os: linux
-  - compiler: gcc
-    os: linux
-    env: PARQUET_BUILD_GROUP=toolchain
     before_script:
-    script:
-    - $TRAVIS_BUILD_DIR/ci/travis_script_static.sh
+    - export PARQUET_CXXFLAGS="-Werror -DARROW_NO_DEPRECATED_API"
+    - source $TRAVIS_BUILD_DIR/ci/before_script_travis.sh
   - compiler: gcc
     os: linux
-    env: PARQUET_BUILD_GROUP=toolchain
     before_script:
-    script:
-    - $TRAVIS_BUILD_DIR/ci/travis_script_toolchain.sh
+    - export PARQUET_CXXFLAGS="-Werror"
+    - source $TRAVIS_BUILD_DIR/ci/before_script_travis.sh
   - compiler: clang
     os: linux
+    before_script:
+    - export PARQUET_CXXFLAGS="-Werror -DARROW_NO_DEPRECATED_API"
+    - source $TRAVIS_BUILD_DIR/ci/before_script_travis.sh
   - compiler: clang
     os: osx
     osx_image: xcode6.4
     addons:
+    before_script:
+    - export PARQUET_CXXFLAGS="-Werror -DARROW_NO_DEPRECATED_API"
+    - source $TRAVIS_BUILD_DIR/ci/before_script_travis.sh
     before_install:
     - mkdir $TRAVIS_BUILD_DIR/parquet-build
     - pushd $TRAVIS_BUILD_DIR/parquet-build
+  - compiler: gcc
+    os: linux
+    env: PARQUET_BUILD_GROUP=toolchain
+    before_script:
```

Comment:
Can you add `export PARQUET_CXXFLAGS="-Werror -DARROW_NO_DEPRECATED_API"` to static linking as well?